src/main/java/org/broadinstitute/hellbender/tools/validation/compare-with-gatk4.sh

```diff
@@ -0,0 +1,94 @@
+if [ -z "$GATK_HOME" ]; then
+  GATK_HOME=`pwd`
+fi
+
+if [ -z "$PICARD_JAR" ]; then
+  echo "ERROR: PICARD_JAR must be set to the Picard jar, you can get it here: http://broadinstitute.github.io/picard/"  1>&2
+  exit 1
+fi
+
+if [ -z "$GATK3_JAR" ]; then
+  echo "ERROR: GATK3_JAR must be set to the GATK 3 jar, you can get it here: https://www.broadinstitute.org/gatk/download/"  1>&2
+  exit 1
+fi
+
+if [ "$#" -ne 3 ]; then
+    echo "Wrong number of parameters, needs: <your.bam> <reference.fasta> <knownSites.vcf>"
+fi
+
+BIN="${GATK_HOME}/build/install/gatk/bin/gatk"
+
+TMP_FILE=${1%.bam}
+FILE_WO_EXTENSION=${TMP_FILE##*/}
+
+
+# Copy the bam
+TEMP_BAM=tmp.${FILE_WO_EXTENSION}.bam
+UNMARKED_BAM=tmp.unmarked.${FILE_WO_EXTENSION}.bam
+CLEAN_BAM=tmp.clean.${FILE_WO_EXTENSION}.bam
+MARKED_BAM=tmp.marked.${FILE_WO_EXTENSION}.bam
+GATK4_BAM=tmp.gatk4.${FILE_WO_EXTENSION}.bam
+RECAL_TABLE=tmp.${FILE_WO_EXTENSION}.recal.table
+
+PICARD_MARKED_BAM=tmp.picard.marked.${FILE_WO_EXTENSION}.bam
+LEGACY_RECAL_TABLE=tmp.legacy.${FILE_WO_EXTENSION}.recal.table
+LEGACY_BAM=tmp.legacy.${FILE_WO_EXTENSION}.bam
+
+cp $1 ${TEMP_BAM}
+echo ${TEMP_BAM}
+
+
+
+echo "################################################"
+echo "###### Undoing MarkDuplicates and BQSR #########"
+echo "################################################"
+# Undo MarkDuplicates
+${BIN} UnmarkDuplicates -I ${TEMP_BAM} -O ${UNMARKED_BAM}
+
+# Undo BQSR
+${BIN} RevertBaseQualityScores -I ${UNMARKED_BAM} -O ${CLEAN_BAM}
+
+
+echo "################################################"
+echo "############## Run GATK 4 tools ################"
+echo "################################################"
+# Run MarkDuplicates
+${BIN} MarkDuplicates -I ${CLEAN_BAM} -O ${MARKED_BAM} --METRICS_FILE=tmp.metrics --VALIDATION_STRINGENCY LENIENT
+# Run BQSR
+${BIN} BaseRecalibrator -I ${MARKED_BAM} -R $2 -knownSites $3 -O ${RECAL_TABLE}
+${BIN} ApplyBQSR -bqsr_recal_file ${RECAL_TABLE} -I ${MARKED_BAM} -O ${GATK4_BAM}
+rm ${RECAL_TABLE}
+
+
+
+echo "################################################"
+echo "########### Running legacy tools ###############"
+echo "################################################"
+# Run Picard MarkDuplicates
+java -jar ${PICARD_JAR} MarkDuplicates I=${CLEAN_BAM} O=${PICARD_MARKED_BAM} METRICS_FILE=tmp.picard.metrics VALIDATION_STRINGENCY=LENIENT
+
+# Run GATK 3 BQSR...
+${BIN} BaseRecalibrator -I ${PICARD_MARKED_BAM} -R $2 -knownSites $3 -O ${LEGACY_RECAL_TABLE}
+${BIN} ApplyBQSR -bqsr_recal_file ${LEGACY_RECAL_TABLE} -I ${PICARD_MARKED_BAM} -O ${LEGACY_BAM}
+rm ${LEGACY_RECAL_TABLE}
+
+echo "################################################"
+echo "############### Run diff tools #################"
+echo "################################################"
+${BIN} CompareDuplicates -I ${LEGACY_BAM} -I2 ${GATK4_BAM}
+
+${BIN} BqsrRecalBasesCompare -I ${LEGACY_BAM} -I2 ${GATK4_BAM}
+
+
+rm ${TEMP_BAM} ${UNMARKED_BAM} ${CLEAN_BAM} ${MARKED_BAM} ${GATK4_BAM} ${PICARD_MARKED_BAM} ${LEGACY_BAM}
```

Comment:
Provide an argument to the script to allow the GATK4/Picard/Legacy bams to be kept rather than deleted.